spec/features/course/assessment/question/duplication_spec.rb

```diff
@@ -19,29 +25,72 @@
 
     context 'As a Course Manager' do
       let(:user) { create(:course_manager, course: course).user }
+      let!(:programming_question) do
+        questions = source_assessment.questions
+
+        questions.where(actable_type: 'Course::Assessment::Question::Programming').first
+      end
+      let!(:non_programming_question) do
+        questions = source_assessment.questions
+
+        questions.where.not(actable_type: 'Course::Assessment::Question::Programming').first
+      end
+      let!(:programming_language) { programming_question.specific.language }
+
+      # Disable the language before the test, and enable it after the test
+      # Direct SQL is used to avoid the readonly limitations
+      before do
+        ActiveRecord::Base.connection.execute(
+          "UPDATE polyglot_languages SET enabled = false WHERE id = #{programming_language.id}"
+        )
+        # Not too sure why an explicit reload is needed, but without it
+        # the changes do not seem to be reflected in the database
+        programming_question.reload
+      end
+
+      after do
+        ActiveRecord::Base.connection.execute(
+          "UPDATE polyglot_languages SET enabled = true WHERE id = #{programming_language.id}"
+        )
+      end
 
       scenario 'I can duplicate a question from one assessment to another' do
```

Comment:
You should create a separate test to test for duplication of programming question with deprecated language, rather than adding it to the existing test.

Now if anything related to deprecated language fails, it will also fail the normal question duplication.

Each test should have a single responsibility and test a single failure point.

Reply:
done